File: hw1/hw1.py
#! /usr/bin/python

import sys

# import sys to use sys.argv
# sys.argv: a list of strings which the command line is used to execute the script
# The files is counted as a argument
# Open and read the arguments ( filenames ) and print the filenames contents  

for i in range(1,len(sys.argv)):
    with open(sys.argv[i], 'r') as f:
	contents = f.read()
	sys.stdout.write(contents)

File: hw1/hw1.pl
#!/usr/bin/perl

use strict;
use warnings;

# strict is a compiler flag that tells perl compiler to change its beheviour in three ways
# warnings is used to catch unsafe code 
# Open and read the arguments (filenames) and print the filenames contents



my $numArgs = @ARGV;
my $argNum;


foreach $argNum(0 .. $#ARGV)
{
  my $fileName = shift @ARGV; 
  open(my $fh, '<', $fileName) or die $!;

  while(my $row = <$fh>) {
      chomp $row;
      print "$row\n";
  }
  close($fh);

}

